contracts/StakingDistributor.sol

```diff
@@ -1,211 +1,200 @@
 // SPDX-License-Identifier: AGPL-3.0
-pragma solidity ^0.7.5;
-
-import "./libraries/SafeERC20.sol";
-import "./libraries/SafeMath.sol";
+pragma solidity ^0.8.10;
+pragma abicoder v2;
 
 import "./interfaces/IERC20.sol";
 import "./interfaces/ITreasury.sol";
-import "./interfaces/IDistributor.sol";
+import "./interfaces/IStaking.sol";
+import "./interfaces/IUniswapV2Pair.sol";
 
 import "./types/OlympusAccessControlled.sol";
 
-contract Distributor is IDistributor, OlympusAccessControlled {
-    /* ========== DEPENDENCIES ========== */
+/// @notice Updated distributor adds the ability to mint and sync
+///         into Uniswap V2-style liquidity pools, removing the
+///         opportunity-cost dilemma of providing liquidity for
+///         OHM, as well as patches a small bug in the staking contract
+///         that pulls forward an amount of the next epoch rewards. Note that
+///         this implementation bases staking reward distributions on
+///         staked supply, whereas previous implementations used total supply.
```

Comment:
This can be clarified to say "bases staking reward distributions on staked and LP supply".